homeassistant/core.py

```diff
@@ -1257,6 +1265,65 @@ def as_dict(self) -> Dict:
             'config_source': self.config_source
         }
 
+    def _update(self, values: Dict, source: str) -> None:
+        """Update the configuration from a dictionary.
+
+        Async friendly.
+        """
+        from homeassistant.config import _set_time_zone
+        self.config_source = source
+        self.latitude = values['latitude']
+        self.longitude = values['longitude']
+        self.elevation = values['elevation']
+        unit_system = values['unit_system']
+        if unit_system == CONF_UNIT_SYSTEM_IMPERIAL:
+            self.units = IMPERIAL_SYSTEM
+        else:
+            self.units = METRIC_SYSTEM
+        self.location_name = values['location_name']
+        _set_time_zone(self.hass, values['time_zone'])
```

Comment:
This method should be moved to this class too?

Reply:
Fixed.